sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/reminders/ReminderAlarmReceiver.kt

```diff
@@ -0,0 +1,207 @@
+ * are permitted provided that the following conditions are met:
+ *
+ * 1.  Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * 2.  Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation and/or
+ * other materials provided with the distribution.
+ *
+ * 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission. No license is granted to the trademarks of
+ * the copyright holders even if such marks are included in this software.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.sagebionetworks.research.sageresearch.reminders
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.TaskStackBuilder
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Context.NOTIFICATION_SERVICE
+import android.content.Intent
+import android.media.RingtoneManager
+import android.net.Uri
+import android.os.Build
+import android.support.v4.app.NotificationCompat
+import org.researchstack.backbone.ui.MainActivity
+import org.sagebionetworks.research.sageresearch_app_sdk.R
+
+import org.slf4j.LoggerFactory
+
+/**
+ * The ReminderAlarmReceiver can be sub-classed and used to easily control showing notifications to the user
+ */
+open class ReminderAlarmReceiver: BroadcastReceiver() {
+
+    private val logger = LoggerFactory.getLogger(ReminderAlarmReceiver::class.java)
+
+    /**
+     * @property notificationChannelId set on Android OS >= 26, uniquely identified your notifications
+     */
+    protected open val notificationChannelId: String
+        get() {
+            return "Sage-Research NotificationChannelId"
+        }
+
+    /**
+     * @property notificationChannelId set on Android OS >= 26, explains your notifications
+     */
+    protected open val notificationChannelTitle: String
```

Comment:
Nit: make channel title/desc a string resource since these are displayed to the user

Reply:
Good point!